ssl/ssl_local.h

```diff
@@ -1799,7 +1808,30 @@ struct ssl_st {
      */
     const struct sigalg_lookup_st **shared_sigalgs;
     size_t shared_sigalgslen;
-};
+} SSL_CONNECTION;
+
+# define SSL_CONNECTION_FROM_SSL_ONLY_int(ssl, c)                       \
```

Comment:
Could these be inline functions rather than macros? This would also give type checking on arguments.

Barring inline functions there is also the option of adding LHASH-style type checking wrapper identity inline functions.

Reply:
There is already _some_ type checking as the passed-in pointer must point to a structure (or union) with a `type` member so IMO inadvertent bugs should be caught. Also the cast should produce strict aliasing warnings if cast from something else than SSL object. So for now I'll keep it as is.